IdentityCore/src/controllers/MSIDSilentController.m

```diff
@@ -121,28 +122,48 @@ - (void)acquireTokenWithRequest:(MSIDSilentTokenRequest *)request
             MSIDTelemetryAPIEvent *telemetryEvent = [self telemetryAPIEvent];
             [telemetryEvent setUserInformation:result.account];
             [telemetryEvent setIsExtendedLifeTimeToken:result.extendedLifeTimeToken ? MSID_TELEMETRY_VALUE_YES : MSID_TELEMETRY_VALUE_NO];
+            if (self.isLocalFallbackMode)
+            {
+                 [telemetryEvent setSsoExtFallBackFlow:1];
+            }
+            
             [self stopTelemetryEvent:telemetryEvent error:error];
 #endif
             self.currentRequest = nil;
             
             completionBlock(result, error);
             return;
         }
-
+        
         self.currentRequest = nil;
-        MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *ssoResult, NSError *ssoError)
+        MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *fallbackResult, NSError *fallbackError)
         {
             // We don't have any meaningful information from fallback controller (edge case of SSO error) so we use the local controller result earlier
-            if (!ssoResult && (ssoError.code == MSIDErrorSSOExtensionUnexpectedError))
+            // In if block, we handle fallback from local controller to Sso Extention controller, and to use the valid error information
+            if (self.controllerType == MSIDSSOExtensionSilentTokenRequestControllerType)
```

Comment:
This is still breaking the abstraction since the controller now knows its type internally